addons/mail/models/mail_channel.py

```diff
@@ -540,21 +540,48 @@ def _channel_message_notifications(self, message):
                 notifications.append([channel.uuid, dict(message_values)])
         return notifications
 
+    @api.model
+    def partner_info(self, all_partners, direct_partners):
+        """
+        Return the information needed by channel to display channel members
+            :param partners: list of res.parner():
+            :param partners: list of res.parner():
+            :returns: a list of {id', 'name', 'email'} for each partner and adds {im_status, out_of_office_message} for direct_partners.
+            :rtype : list(dict)
+        """
+        partner_infos = {partner['id']: partner for partner in all_partners.sudo().read(['id', 'name', 'email'])}
+        # add im _status and out_of_office_message for direct_partners
+        direct_partners_im_status = {partner['id']: partner for partner in direct_partners.sudo().read(['im_status'])}
+        partner_infos.update(direct_partners_im_status)
+        for user in self.env['res.users'].search([('partner_id', 'in', direct_partners.ids), ('out_of_office_message', '!=', False)]):
+            if user.out_of_office_message:
+                partner_infos[user.partner_id.id]['out_of_office_message'] = user.out_of_office_message
+        return partner_infos
+
     @api.multi
-    def channel_info(self, extra_info = False):
+    def channel_info(self, extra_info=False):
         """ Get the informations header for the current channels
             :returns a list of channels values
             :rtype : list(dict)
         """
+        if not self:
+            return []
         channel_infos = []
-
         # all relations partner_channel on those channels
         all_partner_channel = self.env['mail.channel.partner'].search([('channel_id', 'in', self.ids)])
 
         # all partner infos on those channels
-        partner_infos = all_partner_channel.mapped('partner_id').read(['id', 'name', 'email'])
+        channel_dict = {channel.id: channel for channel in self}
+        all_partners = all_partner_channel.mapped('partner_id')
+        direct_channel_partners = all_partner_channel.filtered(lambda pc: channel_dict[pc.channel_id.id].channel_type == 'chat')
+        direct_partners = direct_channel_partners.mapped('partner_id')
+        partner_infos = self.partner_info(all_partners, direct_partners)
+
+        # add last message preview (only used in mobile)
+        addPreview = self._context.get('isMobile', False) or True
```

Comment:
This condition is always `True`, but I think it should add channel preview only when isMobile is `True`.